src/tests/LitePsmJob-integration.t.sol

```diff
@@ -0,0 +1,166 @@
+        deal(gem, pocket, Art * 2 / GemConversionFactor);
+
+        // ilk line must be greater than Art
+        uint256 newLine = Art * 2 * RAY;
+        VatLike(vat).file(ilk, "line", newLine);
+        (Art,,, line,) = VatLike(vat).ilks(ilk);
+
+        // vat.Line() must be greater than vat.debt()
+        uint256 vatLine = VatLike(vat).Line();
+        uint256 vatDebt = VatLike(vat).debt();
+        VatLike(vat).file("Line", vatLine + vatDebt);
+
+        uint256 wad = litePsm.rush();
+        assertTrue(wad != 0);
+        (bool canWork, bytes memory args) = litePsmJob.workable(NET_A);
+        if (canWork){
+            (bytes4 fn) = abi.decode(args, (bytes4));
+            bytes4 encodedSelector = bytes4(abi.encode(litePsm.fill.selector));
+            assertEq(fn, encodedSelector);
+            vm.expectEmit(false, false, false, true);
+            emit Fill(wad);
+            vm.expectEmit(true, false, false, false);
+            emit Work(NET_A);
+            litePsmJob.work(NET_A, args);
+            wad = litePsm.rush();
+            assertEq(wad, 0);
+        }
+    }
+
+     function test_chug() public {
+        // the dai balance of LitePsm must be greater than the urn's art for this ilk
+        (, uint256 art) = VatLike(vat).urns(ilk, address(litePsm));
+        deal(dai, address(litePsm), art + 1); //must be greater than art so we dont have underflow
+        uint256 wad = litePsm.cut();
+        assertTrue(wad != 0);
+        (bool canWork, bytes memory args) = litePsmJob.workable(NET_A);
+        if (canWork){
+            (bytes4 fn) = abi.decode(args, (bytes4));
+            bytes4 encodedSelector = bytes4(abi.encode(litePsm.chug.selector));
+            assertEq(fn, encodedSelector);
+            vm.expectEmit(false, false, false, true);
+            emit Chug(wad);
+            vm.expectEmit(true, false, false, false);
+            emit Work(NET_A);
+            litePsmJob.work(NET_A, args);
+            wad = litePsm.cut();
+            assertEq(wad, 0);
+        }
+    }
+
+    function test_trim() public {
+        (uint256 Art,,, uint256 line,) = VatLike(vat).ilks(ilk);
+         // Art must be greater than ilk line
+        uint256 newLine = (Art / 2)  * RAY;
+        VatLike(vat).file(ilk, "line", newLine);
+        (Art,,, line,) = VatLike(vat).ilks(ilk);
+        // dai balance of LitePsm must be non-zero
+        deal(dai, address(litePsm), Art);
+        // we first call chug() so workable returns trim() instead of chug()
+        if (litePsm.cut() > MILLION_WAD){
+            litePsm.chug();
+        }
+        uint256 wad = litePsm.gush();
+        assertTrue(wad != 0);
+        (bool canWork, bytes memory args) = litePsmJob.workable(NET_A);
+        if (canWork){
+            (bytes4 fn) = abi.decode(args, (bytes4));
+            bytes4 encodedSelector = bytes4(abi.encode(litePsm.trim.selector));
+            assertEq(fn, encodedSelector);
```

Comment:
Same as above:
```suggestion
            (bytes4 fn) = abi.decode(args, (bytes4));
            assertEq(fn, litePsm.trim.selector);
```

Reply:
good catch! addressed in [f05df30](https://github.com/makerdao/dss-cron/pull/25/commits/f05df30d9f7ea524df6d252cdcd68dc6767cead0)